src/node.cc

```diff
@@ -1352,7 +1352,8 @@ class InternalCallbackScope {
  public:
   InternalCallbackScope(Environment* env,
                         Local<Object> object,
-                        const async_context& asyncContext);
+                        const async_context& asyncContext,
+                        bool allow_empty_object = false);
```

Comment:
Perhaps make this a flag rather than a bool so it's more self-describing in line... e.g. 

```C
InternalCallbackScope cb_scope(&env, Local<Object>(), { 0, 0 }, kAllowEmpty);
```

Or some such...